server/events/vcs/gitlab_client.go

```diff
@@ -416,27 +416,37 @@ func (g *GitlabClient) UpdateStatus(logger logging.SimpleLogging, repo models.Re
 
 	retries := 1
 	delay := 2 * time.Second
-	var commit *gitlab.Commit
+	var commitStatuses []*gitlab.CommitStatus
 	var resp *gitlab.Response
 	var err error
 
+	// get the last commit status with the same ref
+	getCommitStatusesOptions := &gitlab.GetCommitStatusesOptions{
```

Comment:
Can we have better handling on this if the function returns multiple results? Forcing the `PerPage` to `1` is pretty ugly.

If we are not expecting more than one result, we should be checking the length and making a decision on what we want to do in that scenario. i.e. using the first result, but at least report a warning that more than one result was returned, or just erroring.

Reply:
Hi @X-Guardian! Sorry i just got back now after a while.

It's actually very possible that the response contain multiple results, in case the pipeline contain multiple job, but all should have identical PipelineID because for each commit there should only one pipeline per ref (gitlab will screaming error when creating new pipeline with same ref for single commit). So, in the end it doesn't matter which job to use.

On contrary, in existing implementation, we sometime mixed between pipeline from different ref. By filtering the status for specific ref, we can prevent this mixing from happened.

Reply:
So if this is the case, rather than making the assumption, the code should verify that.